### ISSUE TITLE
[Using torchattack with hugging face accelerate for Multi-GPU adversarial training]

### ISSUE THREAD

**Issue by HashmatShadab** [Harry24k/adversarial-attacks-pytorch#117]
## Using Hugging Face accelerate for training usually has the following template

from accelerate import Accelerator
accelerator = Accelerator()
device = accelerator.device
model, optimizer, training_dataloader, scheduler = accelerator.prepare(model, optimizer, training_dataloader, scheduler)
attack = torchattacks.PGD(model, eps=8/255, alpha=2/255, steps=4)

for batch in training_dataloader:

> optimizer.zero_grad()
> inputs, targets = batch
> adv_inputs = attack(inputs, targets)
> outputs = model(adv_inputs)
> loss = loss_function(outputs, targets)
> accelerator.backward(loss)
> optimizer.step()
> scheduler.step()

###  Accelerate  handles the device placement, so  model = model.to(device)  or inputs.to(device) is not used.

So in order to do adversarial training, will the above template work? Or is it related to the issue raised in #95 , then should I use set_device(None)?
